tests/integration/test_partitioning_key.py

```diff
@@ -0,0 +1,722 @@
+
+    spark = (
+        SparkSession.builder.appName("PyIceberg integration test")
+        .config("spark.sql.extensions", "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions")
+        .config("spark.sql.catalog.integration", "org.apache.iceberg.spark.SparkCatalog")
+        .config("spark.sql.catalog.integration.catalog-impl", "org.apache.iceberg.rest.RESTCatalog")
+        .config("spark.sql.catalog.integration.uri", "http://localhost:8181")
+        .config("spark.sql.catalog.integration.io-impl", "org.apache.iceberg.aws.s3.S3FileIO")
+        .config("spark.sql.catalog.integration.warehouse", "s3://warehouse/wh/")
+        .config("spark.sql.catalog.integration.s3.endpoint", "http://localhost:9000")
+        .config("spark.sql.catalog.integration.s3.path-style-access", "true")
+        .config("spark.sql.defaultCatalog", "integration")
+        .getOrCreate()
+    )
+
+    return spark
+
+
+TABLE_SCHEMA = Schema(
+    NestedField(field_id=1, name="boolean_field", field_type=BooleanType(), required=False),
+    NestedField(field_id=2, name="string_field", field_type=StringType(), required=False),
+    NestedField(field_id=3, name="string_long_field", field_type=StringType(), required=False),
+    NestedField(field_id=4, name="int_field", field_type=IntegerType(), required=False),
+    NestedField(field_id=5, name="long_field", field_type=LongType(), required=False),
+    NestedField(field_id=6, name="float_field", field_type=FloatType(), required=False),
+    NestedField(field_id=7, name="double_field", field_type=DoubleType(), required=False),
+    NestedField(field_id=8, name="timestamp_field", field_type=TimestampType(), required=False),
+    NestedField(field_id=9, name="timestamptz_field", field_type=TimestamptzType(), required=False),
+    NestedField(field_id=10, name="date_field", field_type=DateType(), required=False),
+    # NestedField(field_id=11, name="time", field_type=TimeType(), required=False),
+    # NestedField(field_id=12, name="uuid", field_type=UuidType(), required=False),
+    NestedField(field_id=11, name="binary_field", field_type=BinaryType(), required=False),
+    NestedField(field_id=12, name="fixed_field", field_type=FixedType(16), required=False),
+    NestedField(field_id=13, name="decimal", field_type=DecimalType(5, 2), required=False),
+)
+
+
+identifier = "default.test_table"
+
+
+@pytest.mark.parametrize(
+    "partition_fields, partition_values, expected_partition_record, expected_hive_partition_path_slice, spark_create_table_sql_for_justification, spark_data_insert_sql_for_justification",
+    [
+        # Identity Transform
+        (
+            [PartitionField(source_id=1, field_id=1001, transform=IdentityTransform(), name="boolean_field")],
+            [False],
+            Record(boolean_field=False),
+            "boolean_field=False",
+            # pyiceberg writes False while spark writes false, so justification (compare expected value with spark behavior) would fail.
```

Comment:
Skip justification (set spark_create_table_sql_for_justification, spark_data_insert_sql_for_justification to None) as it will fail: spark writes hive partition path as 'false' while pyiceberg writes hive partition path as 'False'. Shall we align with spark?